build/common.targets

```diff
@@ -40,7 +40,7 @@
   </PropertyGroup>
 
   <PropertyGroup Condition=" '$(PackProject)' != 'true' ">
-    <NoWarn>$(NoWarn);CS0419;CS1570;CS1572;CS1573;CS1574;CS1584;CS1587;CS1591</NoWarn>
+    <WarningsNotAsErrors>$(WarningsNotAsErrors);CS0419;CS1570;CS1572;CS1573;CS1574;CS1584;CS1587;CS1591</WarningsNotAsErrors>
```

Comment:
If the team decision from the other comment is that we don't think that XMLDoc will help us improve code quality, then this comment should be ignored (and this PR closed without merging). Otherwise, we can consider different wants to deal with the warnings.

12,000+ warnings is a LOT, and then everyone will have to ignore warnings because it'll be impossible to notice when creating a PR is one more is added.  On the other hand, since we use TreatWarningsAsErrors, there aren't really other scenarios where an actual warning will be raised.

Another option is to use inline suppressions to suppress all the warnings from existing violations. That is a LOT of suppressions, but it will make it really obvious when a new PR adds a public API that doesn't have XMLDoc.

Yet another option is to try editorconfig to [change the severity to suggestion](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-options#severity-level). Similar to nuget.config, [code analysis config reads from multiple locations](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files#precedence), so we can keep these XML doc CA codes as warning for packages we publish, and downgrade to suggestion for other projects.